idea/src/org/jetbrains/kotlin/idea/codeInsight/surroundWith/MoveDeclarationsOutHelper.kt

```diff
@@ -52,16 +52,16 @@ object MoveDeclarationsOutHelper {
             val scope = LocalSearchScope(container)
             val lastStatementOffset = statements[statements.size - 1].textRange.endOffset
 
-            for (statement in statements) {
+            statements.forEachIndexed { i, statement ->
                 if (needToDeclareOut(statement, lastStatementOffset, scope)) {
                     if (statement is KtProperty && statement.initializer != null) {
-                        var declaration = createVariableDeclaration(statement, generateDefaultInitializers)
+                        val isLastStatement = i == statements.size - 1
+                        var declaration = createVariableAssignment(statement, generateDefaultInitializers, isLastStatement)
```

Comment:
I don't think this code is structured well enough. You have three functions that do three sets of entirely different things based on a value of a boolean parameter. I think it's clearer to pull the check out to the top level and to avoid introducing the functions altogether.